kahuna/public/js/edits/image-editor.js

```diff
@@ -115,6 +121,74 @@ imageEditor.controller('ImageEditorCtrl', [
             editsService.update(resource, data, image);
         });
     }
+
+    ctrl.collectionError = false;
+
+    ctrl.selectionMode = true;
+
+    ctrl.openCollectionTree = () => {
+        ctrl.addCollection = true;
+
+        if (!ctrl.collections) {
+            collections.getCollections().then(collections => {
+                ctrl.collections = collections.data.children;
```

Comment:
You could memoize this in the API, and then not have to worry about performance here.